comicapi/genericmetadata.py

```diff
@@ -202,24 +212,122 @@ def get_clean_metadata(self, *attributes: str) -> GenericMetadata:
+                new.pop(r_id)
+            getattr(self, cur).extend(new)
+
+        if isinstance(new, set):
+            remove_set = set()
+            for a in getattr(self, cur):
+                for b in new:
+                    if a.casefold() == b.casefold():
+                        remove_set.add(b)
+                        break
+            new.difference_update(remove_set)
+            getattr(self, cur).update(new)
+
+    def assign_original(self, cur: str, new: Any) -> None:
+        if new is not None:
+            if isinstance(new, str) and len(new) == 0:
+                if isinstance(getattr(self, cur), (list, set)):
+                    getattr(self, cur).clear()
+                else:
+                    setattr(self, cur, None)
+            elif isinstance(new, (list, set)) and len(new) == 0:
+                pass
+            else:
+                setattr(self, cur, new)
+
+    def assign_overwrite(self, cur: str, new: Any) -> None:
+        if isinstance(new, (str, list, set)) and len(new) == 0:
+            if isinstance(getattr(self, cur), (list, set)):
+                getattr(self, cur).clear()
+            else:
+                setattr(self, cur, None)
+        else:
+            setattr(self, cur, new)
+
+    def assign_overwrite_no_clear(self, cur: str, new: Any) -> None:
+        if new is not None:
+            if isinstance(new, (str, list, set)) and len(new) > 0:
+                setattr(self, cur, new)
+            elif isinstance(new, (bool, int, float)):
+                setattr(self, cur, new)
+
+    def assign_add_missing(self, cur: str, new: Any) -> None:
+        if new is not None and not getattr(self, cur):
+            setattr(self, cur, new)
+
+    def assign_combine_new(self, cur: str, new: Any) -> None:
+        # Combine lists and sets and prefer 'new' for strings, floats and ints
+        if new is not None:
+            if isinstance(new, (list, set)) and len(new) > 0 and isinstance(getattr(self, cur), (list, set)):
+                # Check for dupes with casefold to replace accented chars
+                self.assign_dedupe_new(new, cur)
+            elif isinstance(new, str) and len(new) > 0:
+                setattr(self, cur, new)
+            elif isinstance(new, (int, float)) and new > 0:
+                setattr(self, cur, new)
+            else:
+                # Only bool left
+                setattr(self, cur, new)
+
+    def assign_combine_cur(self, cur: str, new: Any) -> None:
+        # Combine lists and sets and prefer 'self' for strings, floats and ints
+        if new is not None:
+            if isinstance(new, str) and (len(getattr(self, cur)) == 0 or getattr(self, cur) is None):
+                setattr(self, cur, new)
+            elif isinstance(new, (list, set)) and len(new) > 0 and isinstance(getattr(self, cur), (list, set)):
+                self.assign_dedupe_cur(new, cur)
+            elif isinstance(new, (int, float)) and getattr(self, cur) == 0:
```

Comment:
This won't update anything 90% of the time int attributes do not start-off as 0 they start as `None`